tests/filecheck/parser-printer/unregistered_dialect.mlir

```diff
@@ -6,12 +6,14 @@
     %y = "op_with_res"() {otherattr = #unknown_attr<b2 ...2 [] <<>>>} : () -> (i32)
     %z = "op_with_operands"(%y, %y) : (i32, i32) -> !unknown_type<{[<()>]}>
     "op"() {ab = !unknown_singleton_type} : () -> ()
+    "test.termop"() : () -> ()
```

Comment:
That looks suspicious to me.
If an operation is not registered, we cannot require a terminator, as we don't know if the operation has a `NoTerminator` trait or not.

Reply:
Actually, let's do that in another PR, I need to add `HasMaybeTrait` that does exactly that.

Reply:
hmm, that goes back to a question I had regarding unregistered ops in xDSL?

Reply:
I opened an issue to track this https://github.com/xdslproject/xdsl/issues/1167